<a href="https://colab.research.google.com/github/adamarispadilla/final-project/blob/main/Padilla_HW_4_AssociationRules.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1 Auto Parts

In [ ]:
import pandas as pd                 ##used for data analysis 
import numpy as np                  ##used for mathematical operations on arrays 
##association rule algorithm
from mlxtend.frequent_patterns import apriori, association_rules 


In [ ]:
##Reads txt file that contains data that will be analyzed and converts it into 
##a csv file. Then it gets transformed into a dataframe. 

read_file=pd.read_csv(r'autoparts.txt', sep='\t')
read_file.to_csv(r'autoparts.csv', index=None)
df=pd.read_csv('autoparts.csv')
df.head()        ##shows the first five rows of the dataframe 

SideLight,Tires,HeadLight,WindShield,Engine,Car
0  BreakPad,SideLight,Tires,Wheels,WindShield,Oil...
1  BreakPad,SideLight,Tires,HeadLight,SteeringWhe...
2  BreakPad,Tires,HeadLight,WindShield,SteeringWh...
3  SideLight,Tires,HeadLight,Wheels,WindShield,St...
4  BreakPad,SideLight,Tires,HeadLight,WindShield,...

In [ ]:
##counts missing data 
print(np.count_nonzero(df.isnull()))

0


In [ ]:
#add a column header 'Auto Parts' 
df=pd.read_csv('autoparts.csv', names=['Auto Parts'])
df.head() 

Auto Parts
0    SideLight,Tires,HeadLight,WindShield,Engine,Car
1  BreakPad,SideLight,Tires,Wheels,WindShield,Oil...
2  BreakPad,SideLight,Tires,HeadLight,SteeringWhe...
3  BreakPad,Tires,HeadLight,WindShield,SteeringWh...
4  SideLight,Tires,HeadLight,Wheels,WindShield,St...

In [ ]:
##states the rows and columns of the dataframe
##This dataframe has 25 rows and 1 column
df.shape

(25, 1)

In [ ]:
##Splits the products into a new format 

data=list(df["Auto Parts"].apply(lambda x:x.split(",")))
data

In [ ]:
#Transform data into a logical dataframe. Each column represents an 
##item and each row represent a record or a transaction for one 
##purchase. TRUE if the autopart transaction exists FALSE if not.

from mlxtend.preprocessing import TransactionEncoder
a=TransactionEncoder()
a_data=a.fit(data).transform(data)
df=pd.DataFrame(a_data, columns=a.columns_)

#Turn False & True to 0 & 1 for easy readability
df=df.replace(False,0)
df=df.replace(True,1)

df

BreakPad  Car  Engine  HeadLight  ...  SteeringWheel  Tires  Wheels  WindShield
0          0    1       1          1  ...              0      1       0           1
1          1    1       0          0  ...              0      1       1           1
2          1    0       1          1  ...              1      1       0           0
3          1    1       0          1  ...              1      1       0           1
4          0    1       0          1  ...              1      1       1           1
5          1    1       0          1  ...              0      1       0           1
6          1    0       1          1  ...              1      1       1           1
7          1    1       1          1  ...              0      1       1           1
8          0    1       1          1  ...              0      0       1           0
9          0    0       1          1  ...              1      0       1           1
10         0    1       0          1  ...              0      1       1           1
11         1    0       1          0  ...              1      0       1           1
12         0    0       1          0  ...              0      0       1           1
13         0    1       1          1  ...              0      1       1           1
14         1    0       1          1  ...              1      0       1           0
15         0    1       1          1  ...              1      0       1           1
16         0    1       1          1  ...              1      1       0           1
17         0    0       1          1  ...              1      1       0           0
18         1    1       0          1  ...              1      1       0           1
19         1    1       0          1  ...              0      0       1           1
20         1    0       1          1  ...              0      0       0           1
21         0    1       0          1  ...              1      1       1           1
22         0    1       1          1  ...              0      1       1           1
23         1    1       1          1  ...              0      0       1           1
24         1    0       1          1  ...              0      1       1           1

[25 rows x 10 columns]

In [ ]:
#set a threshold value for the support value (0.2) and calculate the support value.

#The support value can be set at any value you desire.
#use_colnames=True is to convert the integer values back into its respective item names.
df=apriori(df,min_support=0.2, use_colnames=True)
df

support                                           itemsets
0       0.52                                         (BreakPad)
1       0.64                                              (Car)
2       0.68                                           (Engine)
3       0.88                                        (HeadLight)
4       0.64                                              (Oil)
..       ...                                                ...
283     0.20  (WindShield, HeadLight, Oil, Tires, SteeringWh...
284     0.24  (Wheels, SideLight, WindShield, HeadLight, Tires)
285     0.20  (SideLight, WindShield, HeadLight, Car, Tires,...
286     0.20  (SideLight, WindShield, HeadLight, Car, Oil, T...
287     0.20  (Wheels, SideLight, WindShield, HeadLight, Car...

[288 rows x 2 columns]

From above result, we say there are total we got 287 rules with 0.20 minimum support. Minimum support means the number of transactions in which x appears divided by the total number of transactions.

For example, oil has a support of 0.64 which means that out of all the autoparts, oil appears or gets bought 64% of the time. 

In [ ]:
#Let's view our interpretation values using the Association rule function for lift. 
df_ar=association_rules(df,metric="lift", min_threshold=1)
df_ar

antecedents  ... conviction
0            (Oil)  ...   1.097143
1       (BreakPad)  ...   1.170000
2     (WindShield)  ...   1.008000
3       (BreakPad)  ...   1.040000
4            (Car)  ...   1.920000
...            ...  ...        ...
1935  (Tires, Car)  ...   1.165714
1936   (SideLight)  ...   1.052308
1937  (WindShield)  ...   1.050000
1938         (Car)  ...   1.105455
1939       (Tires)  ...   1.105455

[1940 rows x 9 columns]

From above result, we say that the support is 0.36 calculated by dividing the number of transactions containing oil and breakpad by the total number of transactions. The confidence level is 0.56 shows that out of all the transactions that contain breakpads 0.56 contain oil too. The lift 1.08 tells us that breakpad is 1.08 times more likely to be bought by the customer who also buys oil.

In [ ]:
#Let's view our interpretation values using the Association rule function for confidence.
df_ar=association_rules(df,metric="confidence", min_threshold=0.6)
df_ar

antecedents  ... conviction
0                          (BreakPad)  ...   0.832000
1                          (BreakPad)  ...   0.780000
2                          (BreakPad)  ...   1.170000
3                          (BreakPad)  ...   0.910000
4                          (BreakPad)  ...   0.936000
...                               ...  ...        ...
1199  (Wheels, Tires, Car, HeadLight)  ...   2.640000
1200         (Wheels, SideLight, Car)  ...   1.280000
1201       (Wheels, Tires, SideLight)  ...   1.540000
1202       (Wheels, Tires, HeadLight)  ...   1.493333
1203             (Wheels, Tires, Car)  ...   1.680000

[1204 rows x 9 columns]

From above result, let's take a look at the first row items: breakpad and engine. The two items have the support of 0.32, where breakpad is 0.52 and engine is 0.68. confidence is 0.62, this implies that if a person bought brakepads then there is 62% engine is also bought. There are also higly positive correlation as we can see from the positive values of lift and conviction.

2 COMPUTERSTUFF





In [ ]:
import pandas as pd                 ##used for data analysis 
import numpy as np                  ##used for mathematical operations on arrays 
##association rule algorithm
from mlxtend.frequent_patterns import apriori, association_rules 


In [ ]:
##Reads txt file that contains data that will be analyzed and converts it into 
##a csv file. Then it gets transformed into a dataframe. 

read_file=pd.read_csv(r'computerstuff.txt', sep='\t')
read_file.to_csv(r'computerstuff.csv', index=None)
df=pd.read_csv('computerstuff.csv')
df.head()

Keyboard,Laptop,SSD,HDD,Desktop,HDMI,Monitor
0   Processor,Printer,HDD,SSD,Desktop,HDMI,Monitor
1  Keyboard,Processor,Printer,Laptop,Mouse,Monitor
2         Processor,Printer,HDD,Desktop,HDMI,Mouse
3       Keyboard,Processor,Laptop,HDD,HDMI,Monitor
4             Printer,HDD,SSD,Laptop,Monitor,Mouse

In [ ]:
##counts missing data 
print(np.count_nonzero(df.isnull()))

0


In [ ]:
#add a column header 'computer stuff' 
df=pd.read_csv('computerstuff.csv', names=['computer stuff'])
df.head() 

computer stuff
0     Keyboard,Laptop,SSD,HDD,Desktop,HDMI,Monitor
1   Processor,Printer,HDD,SSD,Desktop,HDMI,Monitor
2  Keyboard,Processor,Printer,Laptop,Mouse,Monitor
3         Processor,Printer,HDD,Desktop,HDMI,Mouse
4       Keyboard,Processor,Laptop,HDD,HDMI,Monitor

In [ ]:
##states the rows and columns of the dataframe
##This dataframe has 25 rows and 1 column
df.shape

(25, 1)

In [ ]:
##Splits the products into a new format 
data=list(df["computer stuff"].apply(lambda x:x.split(",")))
data

[['Keyboard', 'Laptop', 'SSD', 'HDD', 'Desktop', 'HDMI', 'Monitor'],
 ['Processor', 'Printer', 'HDD', 'SSD', 'Desktop', 'HDMI', 'Monitor'],
 ['Keyboard', 'Processor', 'Printer', 'Laptop', 'Mouse', 'Monitor'],
 ['Processor', 'Printer', 'HDD', 'Desktop', 'HDMI', 'Mouse'],
 ['Keyboard', 'Processor', 'Laptop', 'HDD', 'HDMI', 'Monitor'],
 ['Printer', 'HDD', 'SSD', 'Laptop', 'Monitor', 'Mouse'],
 ['Keyboard',
  'Printer',
  'Laptop',
  'HDD',
  'Desktop',
  'HDMI',
  'Mouse',
  'Monitor'],
 ['Keyboard',
  'Processor',
  'Printer',
  'Laptop',
  'SSD',
  'Desktop',
  'HDMI',
  'Monitor'],
 ['Keyboard',
  'Processor',
  'Laptop',
  'SSD',
  'Desktop',
  'HDMI',
  'Mouse',
  'Monitor'],
 ['Processor', 'Laptop', 'HDD', 'Monitor', 'HDMI', 'Mouse'],
 ['Keyboard', 'Laptop', 'SSD', 'HDMI', 'Monitor'],
 ['Keyboard',
  'Processor',
  'Printer',
  'Laptop',
  'HDD',
  'Desktop',
  'HDMI',
  'Monitor'],
 ['Processor', 'Printer', 'Laptop', 'SSD', 'Desktop', 'HDMI'],
 ['Keyboard', 'HDD', 'SSD', 'Desktop',

In [ ]:
#Transform data into a logical dataframe. Each column represents an 
##item and each row represent a record or a transaction for one 
##purchase. #TRUE if the autopart transaction exists FALSE if not.

from mlxtend.preprocessing import TransactionEncoder
a=TransactionEncoder()
a_data=a.fit(data).transform(data)
df=pd.DataFrame(a_data, columns=a.columns_)

#Turn False & True to 0 & 1 for easy readability
df=df.replace(False,0)
df=df.replace(True,1)

df

Desktop  HDD  HDMI  Keyboard  ...  Mouse  Printer  Processor  SSD
0         1    1     1         1  ...      0        0          0    1
1         1    1     1         0  ...      0        1          1    1
2         0    0     0         1  ...      1        1          1    0
3         1    1     1         0  ...      1        1          1    0
4         0    1     1         1  ...      0        0          1    0
5         0    1     0         0  ...      1        1          0    1
6         1    1     1         1  ...      1        1          0    0
7         1    0     1         1  ...      0        1          1    1
8         1    0     1         1  ...      1        0          1    1
9         0    1     1         0  ...      1        0          1    0
10        0    0     1         1  ...      0        0          0    1
11        1    1     1         1  ...      0        1          1    0
12        1    0     1         0  ...      0        1          1    1
13        1    1     0         1  ...      0        0          0    1
14        0    0     1         1  ...      1        1          1    1
15        1    1     1         1  ...      1        1          0    0
16        1    1     1         1  ...      1        0          1    0
17        1    1     1         1  ...      0        1          1    0
18        0    1     0         1  ...      0        0          1    1
19        0    1     0         1  ...      1        1          1    0
20        1    1     1         1  ...      1        0          0    1
21        1    1     1         1  ...      0        0          1    1
22        1    1     0         1  ...      1        1          0    1
23        0    1     0         1  ...      1        1          1    1
24        1    0     1         0  ...      1        1          0    1

[25 rows x 10 columns]

In [ ]:
#set a threshold value for the support value (0.2) and calculate the support value.
df=apriori(df,min_support=0.2,use_colnames=True)
df

support                                      itemsets
0       0.64                                     (Desktop)
1       0.72                                         (HDD)
2       0.72                                        (HDMI)
3       0.76                                    (Keyboard)
4       0.64                                      (Laptop)
..       ...                                           ...
226     0.20       (Monitor, HDMI, SSD, Desktop, Keyboard)
227     0.20        (Monitor, HDMI, HDD, Laptop, Keyboard)
228     0.20   (Processor, Monitor, HDD, Laptop, Keyboard)
229     0.20  (Processor, Monitor, HDMI, Laptop, Keyboard)
230     0.20        (Monitor, HDMI, SSD, Laptop, Keyboard)

[231 rows x 2 columns]

From above result, we say there are total we got 
230 rules with 0.20 minimum support. Minimum support means the number of transactions in which x appears divided by the total number of transactions.

For example, keyboard has a support of 0.74 which means that out of all the computer stuff transactions, keyboard appears or gets bought 64% of the time. 

In [ ]:
#Let's view our interpretation values using the Association rule function.
df_ar=association_rules(df,metric="lift", min_threshold=1)
df_ar

antecedents                        consequents  ...  leverage  conviction
0     (Desktop)                              (HDD)  ...    0.0192    1.120000
1         (HDD)                          (Desktop)  ...    0.0192    1.080000
2     (Desktop)                             (HDMI)  ...    0.0992    2.240000
3        (HDMI)                          (Desktop)  ...    0.0992    1.620000
4     (Desktop)                          (Printer)  ...    0.0160    1.066667
..          ...                                ...  ...       ...         ...
953  (Keyboard)       (Laptop, SSD, Monitor, HDMI)  ...    0.0480    1.085714
954       (SSD)  (Laptop, Keyboard, Monitor, HDMI)  ...    0.0080    1.020000
955    (Laptop)     (SSD, Keyboard, Monitor, HDMI)  ...    0.0464    1.105455
956      (HDMI)   (Laptop, SSD, Keyboard, Monitor)  ...    0.0272    1.052308
957   (Monitor)      (Laptop, SSD, Keyboard, HDMI)  ...    0.0480    1.085714

[958 rows x 9 columns]

From above result, we say that the support is 0.48 calculated by dividing the number of transactions containing desktop and hdd by the total number of transactions. The confidence level is 0.75 shows that out of all the transactions that contain desktop 0.75 contain HDD too. The lift 1.04 tells us that HDD is 1.04 times more likely to be bought by the customer who also buys a desktop.

In [ ]:
#Let's view our interpretation values using the Association rule function. 
df_ar=association_rules(df,metric="confidence", min_threshold=0.6)
df_ar

antecedents          consequents  ...  leverage  conviction
0                   (Desktop)                (HDD)  ...    0.0192    1.120000
1                       (HDD)            (Desktop)  ...    0.0192    1.080000
2                   (Desktop)               (HDMI)  ...    0.0992    2.240000
3                      (HDMI)            (Desktop)  ...    0.0992    1.620000
4                   (Desktop)           (Keyboard)  ...   -0.0064    0.960000
..                        ...                  ...  ...       ...         ...
658     (SSD, Keyboard, HDMI)    (Laptop, Monitor)  ...    0.0544    1.680000
659  (Laptop, Keyboard, HDMI)       (SSD, Monitor)  ...    0.0592    1.493333
660       (Laptop, SSD, HDMI)  (Keyboard, Monitor)  ...    0.0208    1.260000
661    (Laptop, SSD, Monitor)     (Keyboard, HDMI)  ...    0.0544    1.680000
662      (SSD, Monitor, HDMI)   (Laptop, Keyboard)  ...    0.0656    1.820000

[663 rows x 9 columns]

From above result, let's take a look at the 3rd row items: desktop and hdmi. The two items have the support of 0.56, where desktop is 0.64 and HDMI is 0.72. confidence is 0.88, this implies that if a person bought DESKTOP then there is 88% HDMI is also bought. There are also higly positive correlation as we can see from the positive values of lift, leverage, and conviction.

3 Mobile stuff

In [ ]:
import pandas as pd                 ##used for data analysis 
import numpy as np                  ##used for mathematical operations on arrays 
##association rules algorithm
from mlxtend.frequent_patterns import apriori, association_rules 

In [ ]:
import pandas as pd
##Reads txt file that contains data that will be analyzed and converts it into 
##a csv file. Then it gets transformed into a dataframe. 
read_file=pd.read_csv(r'mobilestuff.txt', sep='\t')
read_file.to_csv(r'mobilestuff.csv', index=None)
df=pd.read_csv('mobilestuff.csv')
df.head()

Splitter,Processor,HeadPhones,Lences,HomeButton,Charger,Speakers
0  Processor,HeadPhones,Battery,Lences,Charger,Ca...              
1  Processor,Screen,Lences,HomeButton,Charger,Camera              
2  Splitter,HeadPhones,Screen,Battery,HomeButton,...              
3  Splitter,Processor,Screen,Battery,HomeButton,C...              
4  Splitter,HeadPhones,Screen,Lences,Camera,Speakers

In [ ]:
##counts missing data 
print(np.count_nonzero(df.isnull()))

0


In [ ]:
#add a column header 'mobile stuff' 
df=pd.read_csv('mobilestuff.csv', names=['mobile stuff'])
df.head() 

mobile stuff
0  Splitter,Processor,HeadPhones,Lences,HomeButto...
1  Processor,HeadPhones,Battery,Lences,Charger,Ca...
2  Processor,Screen,Lences,HomeButton,Charger,Camera
3  Splitter,HeadPhones,Screen,Battery,HomeButton,...
4  Splitter,Processor,Screen,Battery,HomeButton,C...

In [ ]:
##states the rows and columns of the dataframe
##This dataframe has 25 rows and 1 column
df.shape

(25, 1)

In [ ]:
##Splits the products into a new format 

data=list(df["mobile stuff"].apply(lambda x:x.split(",")))
data
data

[['Splitter',
  'Processor',
  'HeadPhones',
  'Lences',
  'HomeButton',
  'Charger',
  'Speakers'],
 ['Processor',
  'HeadPhones',
  'Battery',
  'Lences',
  'Charger',
  'Camera',
  'Speakers'],
 ['Processor', 'Screen', 'Lences', 'HomeButton', 'Charger', 'Camera'],
 ['Splitter', 'HeadPhones', 'Screen', 'Battery', 'HomeButton', 'Charger'],
 ['Splitter',
  'Processor',
  'Screen',
  'Battery',
  'HomeButton',
  'Charger',
  'Camera'],
 ['Splitter', 'HeadPhones', 'Screen', 'Lences', 'Camera', 'Speakers'],
 ['Splitter',
  'Processor',
  'HeadPhones',
  'Screen',
  'Battery',
  'HomeButton',
  'Lences',
  'Charger'],
 ['Splitter', 'Screen', 'HomeButton', 'Charger', 'Camera'],
 ['Splitter', 'Processor', 'HeadPhones', 'Screen', 'Charger', 'Speakers'],
 ['Processor',
  'HeadPhones',
  'Screen',
  'HomeButton',
  'Lences',
  'Charger',
  'Speakers'],
 ['Splitter', 'Processor', 'HeadPhones', 'Screen', 'Lences', 'Speakers'],
 ['Processor', 'HeadPhones', 'Battery', 'Lences', 'Charger', 'Camera']

In [ ]:
#Transform data into a logical dataframe. 
#TRUE if the transaction occurs FALSE if not.

from mlxtend.preprocessing import TransactionEncoder
a=TransactionEncoder()
a_data=a.fit(data).transform(data)
df=pd.DataFrame(a_data, columns=a.columns_)

#Turn False & True to 0 & 1 for easy readability
df=df.replace(False,0)
df=df.replace(True,1)

df

Battery  Camera  Charger  HeadPhones  ...  Processor  Screen  Speakers  Splitter
0         0       0        1           1  ...          1       0         1         1
1         1       1        1           1  ...          1       0         1         0
2         0       1        1           0  ...          1       1         0         0
3         1       0        1           1  ...          0       1         0         1
4         1       1        1           0  ...          1       1         0         1
5         0       1        0           1  ...          0       1         1         1
6         1       0        1           1  ...          1       1         0         1
7         0       1        1           0  ...          0       1         0         1
8         0       0        1           1  ...          1       1         1         1
9         0       0        1           1  ...          1       1         1         0
10        0       0        0           1  ...          1       1         1         1
11        1       1        1           1  ...          1       0         0         0
12        0       1        1           0  ...          1       1         0         0
13        1       1        1           1  ...          0       0         1         1
14        1       1        0           1  ...          0       1         0         0
15        1       1        1           1  ...          1       1         0         0
16        1       0        1           0  ...          0       1         1         1
17        1       1        1           1  ...          1       1         1         1
18        0       1        1           0  ...          1       1         0         0
19        1       0        1           1  ...          1       0         1         0
20        1       1        0           1  ...          1       1         0         1
21        1       1        1           1  ...          0       0         0         1
22        1       1        1           1  ...          0       1         1         1
23        0       0        1           1  ...          0       1         1         1
24        0       0        1           0  ...          0       0         1         1

[25 rows x 10 columns]

In [ ]:
#set a threshold value for the support value (0.2) and calculate the support value.
#The support value can be set at any value you desire (according to the situation)
#use_colnames=True is to convert the integer values back into its respective item names.
df=apriori(df,min_support=0.2,use_colnames=True)
df

support                                           itemsets
0       0.56                                          (Battery)
1       0.60                                           (Camera)
2       0.84                                          (Charger)
3       0.72                                       (HeadPhones)
4       0.64                                       (HomeButton)
..       ...                                                ...
215     0.20               (Speakers, Lences, Splitter, Screen)
216     0.20     (HeadPhones, Battery, Lences, Camera, Charger)
217     0.20  (HeadPhones, Processor, Battery, Lences, Charger)
218     0.20   (HomeButton, Processor, Camera, Charger, Screen)
219     0.20  (HeadPhones, Processor, Speakers, Lences, Char...

[220 rows x 2 columns]

From above result, we say there are total we got 219 rules with 0.20 minimum support. Minimum support means the number of transactions in which x appears divided by the total number of transactions.

For example, homebuttom has a support of 0.64 which means that out of all the mobile stuff transactions, keyboard appears or gets bought 64% of the time.

In [ ]:
#Let's view our interpretation values using the Association rule function.
df_ar=association_rules(df,metric="lift", min_threshold=1)
df_ar

antecedents  ... conviction
0                  (Battery)  ...   1.400000
1                   (Camera)  ...   1.320000
2                  (Battery)  ...   1.120000
3                  (Charger)  ...   1.026667
4                  (Battery)  ...   1.960000
..                       ...  ...        ...
981  (HeadPhones, Processor)  ...   1.246667
982               (Speakers)  ...   1.170000
983                 (Lences)  ...   1.076667
984              (Processor)  ...   1.140000
985             (HeadPhones)  ...   1.107692

[986 rows x 9 columns]

From above result, we say that the support is 0.48 calculated by dividing the number of transactions containing battery and headphones by the total number of transactions. The confidence level is 0.86 shows that out of all the transactions that contain battery 0.86 contain headphones too. The lift 1.19 tells us that headphone is 1.19 times more likely to be bought by the customer who also buys a battery.

In [ ]:
#Let's view our interpretation values using the Association rule function.
df_ar=association_rules(df,metric="confidence", min_threshold=0.6)
df_ar

antecedents  ... conviction
0                            (Battery)  ...       1.40
1                             (Camera)  ...       1.32
2                            (Battery)  ...       1.12
3                            (Battery)  ...       1.96
4                         (HeadPhones)  ...       1.32
..                                 ...  ...        ...
584     (Processor, Speakers, Charger)  ...       2.88
585      (Speakers, Lences, Processor)  ...       2.64
586     (Speakers, Lences, HeadPhones)  ...       1.28
587  (HeadPhones, Speakers, Processor)  ...       1.68
588              (Speakers, Processor)  ...       2.24

[589 rows x 9 columns]

From above result, let's take a look at the first row items: battery and camera. The two items have the support of 0.40, where battery is 0.56 and camera is 0.60. confidence is 0.71, this implies that if a person bought battery then there is 71% camera is also bought. There are also higly positive correlation as we can see from the positive values of lift, leverage, and conviction.

3) Based on your analysis and results, consider these questions: 

Your boss wants to use some coupons to increase the sales on those items that are selling well, what advice can you give to your boss based on your analysis? how about those items that are NOT selling well? 



---




For items that are selling well, I would tell my boss to select items that have a high support number because those have a higher rate of being purchased.For items not selling well, I would tell my boss to select items that have a low support number because those have a lower rate of being purchased. Maybe a coupon would incentives people to buy it more.



On certain transactions, you see a certain purchasing pattern. If you know the customer profile (age, zipcode, gender, etc...), Can you do a specific marketing on those customer groups? how?


---


Yes you cultivate specific marketing plan on certain customer groups by pairing up the item with things that align with the customers profile. For example if the item is a headphone and the customer profile is a teen with the age 12-18, then you can post a short advertisment on tiktok since most teens use that platform.